src/sema/assembly/expression.rs

```diff
@@ -640,7 +639,101 @@ fn resolve_member_access(
 
 /// Check if an assembly expression has been used correctly in a assignment or if the member access
 /// has a valid expression given the context.
-pub(crate) fn check_type(expr: &AssemblyExpression, context: &ExprContext) -> Option<Diagnostic> {
+pub(crate) fn check_type(
+    expr: &AssemblyExpression,
+    context: &ExprContext,
+    ns: &mut Namespace,
+    symtable: &mut Symtable,
+) -> Option<Diagnostic> {
+    if context.lvalue {
+        match expr {
+            AssemblyExpression::SolidityLocalVariable(
+                _,
+                _,
+                Some(StorageLocation::Storage(_)),
+                ..,
+            )
+            | AssemblyExpression::StorageVariable(..) => {
+                return Some(Diagnostic::error(
+                    expr.loc(),
+                    "storage variables cannot be assigned any value in assembly. You may use \"sstore()\"".to_string()
```

Comment:
We have inconsistent quoting in solang error messages. I think we should always use ‘foo’. However some messages use 'foo' (which should be fixed) and now this introduces "foo".